### DISCUSSION TITLE
No admin token set for new admin user

### DISCUSSION THREAD

**Discussion by MayankPunghal** [nz-m/SocialEcho#37] [Q&A] [answered]
I created a new user with Admin rights using the bash file. But it gives me 401 unauthorized when I try to login on admin portal with the admin credentials. I am not sure what I am doing wrong here. Can you suggest what the issue might be, because I have set up the env variables and the DB correctly. Everything else is working fine, even the user login and the home screen flow is working fine.

Although there is one thing that I noticed.
In user signing the url that gets formed is : http://localhost:3000/users/signin
where as in the case of admin it is : http://localhost:3000/admin/undefined/admin/signin

I also checked the DB and the AdminToken is empty as even though I ahve multiple users with Admin rights created already.

**Comment by nz-m** [comment]
There is no route at **/users/signin**. The main sign in page should be accessible at **http://localhost:3000/signin**, and there is a link to the admin login, which will allow you to go to the admin sign in page at **http://localhost:3000/admin/signin**.

**Reply by MayankPunghal** [reply]
No, those are not the page routes those are the url that form for the API on submit click. Also can you please tell me about the Unauthorized alert on Admin login. Does it not create a token for Admin on creating a new admin user from the bash file?

**Reply by nz-m** [reply]
_No, those are not the page routes those are the url that form for the API on submit click_
In that case, you didn't set up the environment variable in the client.

_Does it not create a token for Admin on creating a new admin user from the bash file?_
A token will be generated after a successful sign-in.

**Reply by MayankPunghal** [reply]
I did set up the API URL in client .env
![image](https://github.com/nz-m/SocialEcho/assets/50830003/ee59dbef-7e81-4aa8-82da-8905701a242c)

All my APIs are working except Admin Sign-in (that API is returning 401 : Unauthorized). Rest of the app is working fine.

And as for the token I was asking about that because I am getting Unauthorized error on ligging in with Admin credentials on Admin Sign-in page.

I am new to React/Redux : )

**Reply by nz-m** [reply, marked as answer]
Backend is running on port 4000, so how is the request being sent to port 3000 (http://localhost:3000/users/signin)? There is an issue with the env file. Try restarting the client.

**Reply by MayankPunghal** [reply]
Haha such a trivial issue. That fixed the problem. I had edited the .env andnot restarted my client just the server. Thank you : )

Also if anyone wants to edit Admin user details will we have to do it manually in Mongo? Or is there a Super Admin kinda thing? Maybe on UI or in the bash file?

**Reply by nz-m** [reply]
There is currently no feature for editing admin user details.
